src/pack.c

```diff
@@ -1141,9 +1144,11 @@ int git_packfile_alloc(struct git_pack_file **pack_out, const char *path)
 	if (git__suffixcmp(path, ".idx") == 0) {
 		size_t root_len = path_len - strlen(".idx");
 
-		memcpy(p->pack_name + root_len, ".keep", sizeof(".keep"));
-		if (git_path_exists(p->pack_name) == true)
-			p->pack_keep = 1;
+		if (!git_ignore_pack_keep_file_check) {
+			memcpy(p->pack_name + root_len, ".keep", sizeof(".keep"));
+			if (git_path_exists(p->pack_name) == true)
+			    p->pack_keep = 1;
```

Comment:
I just noticed that this is "dead code": we don't do anything with that state after setting it, as AFAIK we don't have any API to delete things yet.

Reply:
I retained the code just in case you are writing a pack file, this setting is part of the pack file metadata and would get written to filesystem.
The default git client might/could look into this instead of looking for '.keep' file. This makes it more self contained too.